injector/src/main/java/edu/ucr/cs/riple/injector/changes/AddTypeUseMarkerAnnotation.java

```diff
@@ -0,0 +1,146 @@
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package edu.ucr.cs.riple.injector.changes;
+
+import com.github.javaparser.Range;
+import com.github.javaparser.ast.expr.AnnotationExpr;
+import com.github.javaparser.ast.expr.MarkerAnnotationExpr;
+import com.github.javaparser.ast.nodeTypes.NodeWithAnnotations;
+import com.github.javaparser.ast.nodeTypes.NodeWithRange;
+import com.github.javaparser.ast.type.ArrayType;
+import com.github.javaparser.ast.type.ClassOrInterfaceType;
+import com.github.javaparser.ast.type.PrimitiveType;
+import com.github.javaparser.ast.type.Type;
+import edu.ucr.cs.riple.injector.Helper;
+import edu.ucr.cs.riple.injector.location.Location;
+import edu.ucr.cs.riple.injector.modifications.Insertion;
+import edu.ucr.cs.riple.injector.modifications.Modification;
+import edu.ucr.cs.riple.injector.modifications.MultiPositionModification;
+import java.util.HashSet;
+import java.util.Set;
+import javax.annotation.Nullable;
+
+/**
+ * Adds a type-use marker annotation to a node in the source code. It will add the annotation to
+ * both the declaration and all type arguments.
+ *
+ * <p>For instance, for the node: {@code java.util.Map<String, java.lang.String> list;} It will add
+ * the {@code @Nullable} annotation to the type {@code Map} and all the type arguments. The final
+ * output will be: {@code java.util.@Nullable Map<@Nullable String, java.lang.@Nullable String>
+ * list;}
+ */
+public class AddTypeUseMarkerAnnotation extends AddMarkerAnnotation {
+
+  public AddTypeUseMarkerAnnotation(Location location, String annotation) {
+    super(location, annotation);
+  }
+
+  @Override
+  @Nullable
+  public <T extends NodeWithAnnotations<?> & NodeWithRange<?>>
+      Modification computeTextModificationOn(T node) {
+    if (node.getRange().isEmpty()) {
+      return null;
+    }
+
+    AnnotationExpr annotationExpr = new MarkerAnnotationExpr(annotationName.simpleName);
+
+    Set<Modification> modifications = new HashSet<>();
+    Type type = Helper.getType(node);
+
+    if (!Helper.isAnnotatedWith(type, annotationExpr)) {
+      // Annotate the type.
+      Range range = findTypeRange(type);
+      if (range == null) {
+        return null;
+      }
+      modifications.add(new Insertion(annotationExpr.toString(), range.begin));
+    }
+    // Annotate type arguments. Note: We currently do not annotate Map<K, V>[] as @Annot Map<@Annot
+    // K, @Annot V>[] and will leave it as @Annot Map<K, V>[]
```

Comment:
Why do we have this special case?

Reply:
I thought maybe we can add this support once we see the use-case. But I think it worths to just add the support here anyway since it is very straightforward. f1b86c9e6bae5ea1c41a7c1a4700f47c9490eb6b
Also updated the test case.